dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java

```diff
@@ -228,143 +230,192 @@ private Container getWorkingContainerById(final String containerIdOrPath, final
-                    && APILocator.getPortletAPI().hasContainerManagerRights(user);
-            final boolean hasReadPermissionOnContainer = permissionAPI.doesUserHavePermission(container, PERMISSION_READ, user, false);
-            contextMap.put("EDIT_CONTAINER_PERMISSION" + container.getIdentifier(), hasWritePermissionOnContainer);
-            if (Config.getBooleanProperty("SIMPLE_PAGE_CONTENT_PERMISSIONING", true)) {
-                contextMap.put("USE_CONTAINER_PERMISSION" + container.getIdentifier(), true);
-            } else {
-                contextMap.put("USE_CONTAINER_PERMISSION" + container.getIdentifier(), hasReadPermissionOnContainer);
-            }
+            this.addPermissions(container);
 
-            final Map<String, List<Map<String,Object>>> contentMaps = Maps.newLinkedHashMap();
-            final Map<String, List<String>> containerUuidPersona = Maps.newHashMap();
+            final Map<String, List<Map<String, Object>>> contentMaps = Maps.newLinkedHashMap();
+            final Map<String, List<String>> containerUuidPersona     = Maps.newHashMap();
             for (final String uniqueId : pageContents.row(containerId).keySet()) {
 
                 final String uniqueUUIDForRender = needParseContainerPrefix(container, uniqueId) ?
-                                ParseContainer.getDotParserContainerUUID(uniqueId) :
-                                uniqueId;
+                        ParseContainer.getDotParserContainerUUID(uniqueId) : uniqueId;
 
-                if(ContainerUUID.UUID_DEFAULT_VALUE.equals(uniqueId)) {
+                if (ContainerUUID.UUID_DEFAULT_VALUE.equals(uniqueId)) {
                     continue;
                 }
 
                 final Collection<PersonalizedContentlet> personalizedContentletSet = pageContents.get(containerId, uniqueId);
-                final List<Map<String, Object>> personalizedContentletMap = Lists.newArrayList();
+                final List<Map<String, Object>> personalizedContentletMap          = Lists.newArrayList();
+                int   contentletIncludedCount = 1;
+                for (final PersonalizedContentlet personalizedContentlet : personalizedContentletSet) {
 
-                for (final PersonalizedContentlet  personalizedContentlet : personalizedContentletSet) {
-                    final Contentlet contentlet = getContentlet(personalizedContentlet);
+                    final Contentlet contentlet = this.getContentlet(personalizedContentlet);
 
                     if (contentlet == null) {
+
+                        continue;
+                    }
+
+                    if (container.getMaxContentlets() < contentletIncludedCount) {
+
+                        Logger.debug(this, ()-> "Contentlet: "          + contentlet.getIdentifier()
+                                + ", has been skipped. Max contentlet: "    + container.getMaxContentlets()
+                                + ", has been overcome for the container: " + containerId);
```

Comment:
I think this case just could happend if the user change the Max Contentlet to a number less than the contentlets currently added, maybe in this case we should sent a message to the user (if the user is a backend user)